src/main/java/com/gamzabat/algohub/feature/recommendation/domain/GroupActivityWeekly.java

```diff
@@ -0,0 +1,51 @@
+package com.gamzabat.algohub.feature.recommendation.domain;
+
+import java.time.LocalDateTime;
+
+import com.gamzabat.algohub.feature.group.studygroup.domain.StudyGroup;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name = "group_activity_weekly")
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
```

Comment:
지금 다른 엔티티들에는 추가가 안되어있긴한데.. 아래처럼 constructor에 검증을 추가하는건 어떨까요??
전에 builder 쓰다가 필드를 하나씩 빼먹는 실수가 생겼었는데, 아래처럼 검증 추가하니 좀 낫더라구요..ㅎㅎ

```suggestion
@Builder
public GroupActivityWeekly(
        Long id,
        @NotNull StudyGroup studyGroup,
        @NotNull LocalDateTime weekStart,
        @NotNull LocalDateTime weekEnd,
        @NotNull Integer submissions,
        @NotNull Integer comments,
        @NotNull Double activeScore
) {
    this.id = id;
    this.studyGroup = studyGroup;
    this.weekStart = weekStart;
    this.weekEnd = weekEnd;
    this.submissions = submissions;
    this.comments = comments;
    this.activeScore = activeScore;
}
```

부가적인 부분이라 적용은 안하셔도 괜찮습니다!

Reply:
null 값이 들어가거나 나중에 save() 단계에서 NPE가 터질 수도 있겠네요 ㅎㅎ

생성자 레벨에서 @Builder를 사용하고 @NotNull로 명확하게 검증 로직을 추가하는 방식이 훨씬 더 안정적이고 견고한 코드가 될 것 같아서 GroupActivityWeekly 와 GroupJoinMonthlyRolling 등 다른 엔티티들에도 동일하게 적용해두겠습니다 :)

좋은 제안 감사드립니다 :)